hammr/commands/image/image.py

```diff
@@ -271,117 +270,142 @@ def help_deploy(self):
-                        imgStatus = self.get_image_status(image.status)
-                        table.add_row([image.dbId, image.name, image.version, image.revision, image.targetFormat.name,
-                                       image.created.strftime("%Y-%m-%d %H:%M:%S"), size(image.size),
-                                       "X" if image.compress else "", imgStatus])
-                        deleteImage = image
-                if deleteImage is not None:
-                    print table.draw() + "\n"
-                    if generics_utils.query_yes_no(
-                                    "Do you really want to delete image with id " + str(deleteImage.dbId)):
-                        self.api.Users(self.login).Appliances(
-                            generics_utils.extract_id(deleteImage.applianceUri)).Images(deleteImage.dbId).Delete()
-                        printer.out("Image deleted", printer.OK)
-                else:
-                    printer.out("Image not found", printer.ERROR)
+            printer.out("Searching image with id [" + do_args.id + "] ...")
+            images = self.get_all_images()
+            if images is None :
+                raise ValueError("No image found")
+
+            table = Texttable(800)
+            table.set_cols_dtype(["t", "t", "t", "t", "t", "t", "t", "t", "t"])
+            table.header(["Id", "Name", "Version", "Rev.", "Format", "Created", "Size", "Compressed", "Status"])
+            delete_image = None
+            for image in images:
+                if str(image.dbId) == str(do_args.id):
+                    img_status = self.get_image_status(image.status)
+                    table.add_row([image.dbId, image.name, image.version, image.revision, image.targetFormat.name,
+                                   image.created.strftime("%Y-%m-%d %H:%M:%S"), size(image.size),
+                                   "X" if image.compress else "", img_status])
+                    delete_image = image
+            if delete_image is not None:
+                print table.draw() + "\n"
+                if generics_utils.query_yes_no(
+                                "Do you really want to delete image with id " + str(delete_image.dbId)):
+
+                    if is_uri_based_on_appliance(delete_image.uri):
+                        appliance_id = extract_appliance_id(delete_image.uri)
+                        self.api.Users(self.login).Appliances(appliance_id).Images(delete_image.dbId).Delete()
 
+                    elif is_uri_based_on_scan(delete_image.uri):
+                        scanned_instance_id = extract_scannedinstance_id(delete_image.uri)
+                        scan_id = extract_scan_id(delete_image.uri)
+                        self.api.Users(self.login).Scannedinstances(scanned_instance_id).Scans(scan_id).Images(None, delete_image.dbId).Delete()
+
+                    else :
+                        raise ValueError("Image with id '" + do_args.id + " not found.")
```

Comment:
I think this error message is not appropriate (the image has been found but there is an internal problem and its uri can not be interpreted)